lib/client/okta.go

```diff
@@ -286,33 +290,53 @@ func (o *OktaClient) GetSessionToken() string {
 
 // Will prompt the user to select one of the configured MFA devices if an MFA
 // configuration isn't provided.
-func (o *OktaClient) selectMFADevice() (*oktaUserAuthnFactor, error) {
+func (o *OktaClient) selectMFADevice() (mfa.Device, error) {
 	var haveMFAConfig bool
+	var availableDevices []mfa.Device
+	var availableFactors []types.OktaUserAuthnFactor
+
 	factors := o.userAuth.Embedded.Factors
 
 	haveMFAConfig = o.creds.MFA.Provider != "" && o.creds.MFA.FactorType != ""
 
+	for _, factor := range factors {
+		supported := false
+		var device mfa.Device
+		for _, dev := range o.mfaDevices {
```

Comment:
I'm not sure about this `Supported` interface. Looks like you're just doing a `switch/case` with extra steps?

Having a look at the implementations, the inputs they use to make the decision are only the factor type and the provider, right? So no need to expose them to the whole `OktaUserAuthnFactor` struct

Reply:
It is essentially just a switch case but by using `Supported` we can allow external/custom implementations for `mfa.Device`. I've added two public methods to `OktaClient` to support this. 

`RegisterMFADeviceType` and `ClearMFADevices`

Reply:
I've updated the logic for this a bit more, is the reason/value clearer now?